README.md

```diff
@@ -1,526 +1,917 @@
+
+Further extending the `@logged` decorator, we can make it handle props as well, logging when the prop is initialized and whenever it is accessed:
+
+```js
+function logged(value, { kind, name }) {
+  if (kind === "prop") {
+    let { get, set } = value;
+
+    return {
+      get() {
+        console.log(`getting ${name}`);
+
+        return get.call(this);
+      },
+
+      set(val) {
+        console.log(`setting ${name} to ${val}`);
+
+        return set.call(this, val);
+      },
+
+      initialize(initialValue) {
+        console.log(`initializing ${name} with value ${initialValue}`);
+        return initialValue;
+      }
+    };
+  }
 
-export function setBox(box, contents) {
-  return key.set(box, contents);
+  // ...
 }
 
-export function getBox(box) {
-  return key.get(box);
+class C {
+  @logged prop x = 1;
 }
+
+let c = new C();
+// initializing x with value 1
+c.x;
+// getting x
+c.x = 123;
+// setting x to 123
 ```
 
-Note that this is a bit of a hack, and could be done better with constructs like references to private names with [`private.name`](https://gist.github.com/littledan/ab73ff08f98f33088a0072ad202445b1) and broader scope of private names with [`private`/`with`](https://gist.github.com/littledan/5451d6426a8ed65c0f3c2822c51314d1). But it shows that this decorator proposal "naturally" exposes existing things in a useful way.
+This example roughly "desugars" to the following:
+
+```js
+class C {
+  #x = initializeX(1);
 
-```mjs
-// private-key.mjs
-export class PrivateKey {
-  #get;
-  #set;
-  
-  show({get, set}) {
-    assert(this.#get === undefined && this.#set === undefined);
-    this.#get = get;
-    this.#set = set;
-    return {get, set};
+  get x() {
+    return this.#x;
   }
-  get(obj) {
-    return this.#get.call(obj);
+
+  set x(val) {
+    this.#x = val;
   }
-  set(obj, value) {
-    return this.#set.call(obj, value);
+}
+
+let { get: oldGet, set: oldSet } = Object.getOwnPropertyDescriptor(C.prototype, "x");
+
+let {
+  get: newGet,
+  set: newSet,
+  initialize: initializeX
+} = logged(
+  { get: oldGet, set: oldSet },
+  {
+    kind: "prop",
+    name: "x",
+    isStatic: false,
+    isPrivate: false,
+    defineMetadata() { /**/ }
   }
+);
+
+Object.defineProperty(C.prototype, "x", { get: newGet, set: newSet });
+```
+
+#### Class Initialized Methods
+
+Class initialized methods are a new construct, defined by adding the `init` keyword in front of a class method:
```

Comment:
What are the use cases for this in the absence of a decorator?

Reply:
Similar to the `prop` keyword, the main point is to change the semantics of methods so that decorators can layer naturally on top, replacing a `init-method` with another `init-method`. This is important for use cases such as `@bound`, where some initialization logic needs to run with the method. 

Doing it this way keeps it consistent with the strategy for `prop`, so we don't have two different ways of introducing alternative semantics for decoratable values. @rbuckton also thought that this made more sense than `@init:`, because it does change the semantics of the element.

I agree though that the potential use cases for `init` on its own are even more limited than the use cases for `prop` on its own.

Reply:
Changing the semantics in a decorator is fine tho, the issue was about not being able to _statically know_ how the semantics were being changed.